### PR TITLE
normalize-memrefs: Normalize memref.alloca

### PR DESCRIPTION
**Pull request by mgehre-amd**
The pass was only handling memref.alloc,
and this extends it to also handle memref.alloca.

### CODE REVIEW

File: mlir/lib/Dialect/MemRef/Transforms/NormalizeMemRefs.cpp [hunk 1]
```diff
@@ -347,6 +358,11 @@ void NormalizeMemRefs::normalizeFuncOpMemRefs(func::FuncOp funcOp,
   for (memref::AllocOp allocOp : allocOps)
     (void)normalizeMemRef(&allocOp);
 
+  SmallVector<memref::AllocaOp, 4> allocaOps;
```
Comment (cferry-AMD) [inline mlir/lib/Dialect/MemRef/Transforms/NormalizeMemRefs.cpp:361]:
```suggestion
  SmallVector<memref::AllocaOp> allocaOps;
```

Perhaps also do the same above ? :)

File: mlir/test/Dialect/MemRef/normalize-memrefs.mlir [hunk 2]
```diff
@@ -27,6 +27,20 @@ func.func @permute() {
...
 
+// CHECK-LABEL: func @alloca
+func.func @alloca(%idx : index) {
+  // CHECK-NEXT: memref.alloca() : memref<65xf32>
+  %A = memref.alloca() : memref<64xf32, affine_map<(d0) -> (d0 + 1)>>
+  // CHECK-NEXT: affine.load %{{.*}}[symbol(%arg0) + 1] : memref<65xf32>
+  affine.load %A[%idx] : memref<64xf32, affine_map<(d0) -> (d0 + 1)>>
+  affine.for %i = 0 to 64 {
+    %1 = affine.load %A[%i] : memref<64xf32, affine_map<(d0) -> (d0 + 1)>>
+    "prevent.dce"(%1) : (f32) -> ()
```
Comment (cferry-AMD) [inline mlir/test/Dialect/MemRef/normalize-memrefs.mlir:38]: Thanks for the DCE prevention tip!

**Review by cferry-AMD — APPROVED** [review]
LGTM

File: mlir/lib/Dialect/MemRef/Transforms/NormalizeMemRefs.cpp [hunk 3]
```diff
@@ -347,6 +358,11 @@ void NormalizeMemRefs::normalizeFuncOpMemRefs(func::FuncOp funcOp,
   for (memref::AllocOp allocOp : allocOps)
     (void)normalizeMemRef(&allocOp);
 
+  SmallVector<memref::AllocaOp> allocaOps;
```
Comment (TinaAMD) [inline mlir/lib/Dialect/MemRef/Transforms/NormalizeMemRefs.cpp:361]: Can you add `AllocaOp` to the documentation of the function (also to the `normalizable...`)?

**Review by TinaAMD — APPROVED** [review]
LGTM

**Pull request merged**